homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,273 @@
+    identifier: str
+    name: str
+    current_version: str
+    available_version: str
+    changelog_content: str | None = None
+    changelog_url: str | None = None
+    icon_url: str | None = None
+    supports_backup: bool = False
+
+
+class UpdateManager:
+    """Update manager for the update integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the update manager."""
+        self._hass = hass
+        self._store = storage.Store(
+            hass=hass,
+            version=STORAGE_VERSION,
+            key=DOMAIN,
+        )
+        self._skip: set[str] = set()
+        self._platforms: dict[str, UpdatePlatformProtocol] = {}
+        self._loaded = False
+
+    async def add_platform(
+        self,
+        hass: HomeAssistant,
+        integration_domain: str,
+        platform: UpdatePlatformProtocol,
+    ) -> None:
+        """Add a platform to the update manager."""
+        self._platforms[integration_domain] = platform
+
+    async def _load(self) -> None:
+        """Load platforms and data from storage."""
+        await integration_platform.async_process_integration_platforms(
+            self._hass, DOMAIN, self.add_platform
+        )
+        from_storage = await self._store.async_load()
+        if isinstance(from_storage, dict):
+            self._skip = set(from_storage["skipped"])
+
+        self._loaded = True
+
+    async def gather_updates(self) -> list[dict[str, Any]]:
+        """Gather updates."""
+        if not self._loaded:
+            await self._load()
+
+        updates: dict[str, list[UpdateDescription] | None] = {}
+
+        for domain, update_descriptions in zip(
+            self._platforms,
+            await asyncio.gather(
+                *(
+                    self._get_integration_info(integration_domain, registration)
+                    for integration_domain, registration in self._platforms.items()
+                )
+            ),
+        ):
+            updates[domain] = update_descriptions
+
+        return [
+            {
+                "domain": integration_domain,
+                **dataclasses.asdict(description),
+            }
+            for integration_domain, update_descriptions in updates.items()
+            if update_descriptions is not None
+            for description in update_descriptions
+            if f"{integration_domain}_{description.identifier}_{description.available_version}"
+            not in self._skip
+        ]
+
+    async def domain_is_valid(self, domain: str) -> bool:
+        """Return if the domain is valid."""
+        if not self._loaded:
+            await self._load()
+        return domain in self._platforms
+
+    @callback
+    def _data_to_save(self) -> dict[str, Any]:
+        """Schedule storing the data."""
+        return {"skipped": list(self._skip)}
+
+    async def perform_update(
+        self,
+        domain: str,
+        identifier: str,
+        version: str,
+        **kwargs: Any,
```

Comment:
When will the arbitrary keyword arguments be used?

Reply:
for `backup` and whatever we decide to add in the future.